File: src/main/java/org/books/api/OwnerController.java
package org.books.api;

import org.books.beans.LoggedOwner;
import org.books.entities.Owner;
import org.books.services.OwnerService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping(value = "/api/public", produces = "application/json")
public class OwnerController {

    @Autowired
    private OwnerService ownerService;

    @PostMapping(value = "/register", consumes = "application/json")
    public ResponseEntity<LoggedOwner> register(@RequestBody Owner owner) {
        List<Owner> possibleOwners = ownerService.procurarOwner(owner);

        if (!possibleOwners.isEmpty()) {
            return ResponseEntity
                .badRequest()
                .build();
        }

        return ResponseEntity
            .ok(ownerService.criarOwner(owner));
    }

}

File: src/main/java/org/books/api/OwnerController.kt
package org.books.api

import org.books.beans.LoggedOwner
import org.books.entities.Owner
import org.books.services.OwnerService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = "/api/public", produces = arrayOf("application/json"))
class OwnerController {

    @Autowired
    private val ownerService: OwnerService? = null

    @PostMapping(value = "/register", consumes = arrayOf("application/json"))
    fun register(@RequestBody owner: Owner): ResponseEntity<LoggedOwner> {
        val possibleOwners = ownerService!!.procurarOwner(owner)

        return if (!possibleOwners.isEmpty()) {
            ResponseEntity
                    .badRequest()
                    .build()
        } else ResponseEntity
                .ok(ownerService.criarOwner(owner))

    }

}
